src/Type.zig

```diff
@@ -162,24 +162,70 @@ pub const Enum = struct {
         return e;
     }
 };
+// might not need all 4 of these when finished,
+// but currently it helps havaing all 4 when diff-ing
+// the reust code.
+pub const TypeLayout = struct {
+    /// The size of the type in bits.
+    ///
+    /// This is the value returned by `sizeof` and C and `std::mem::size_of` in Rust
+    /// (but in bits instead of bytes). This is a multiple of `pointer_alignment_bits`.
+    size_bits: u29,
```

Comment:
It's weird that this uses `u29` when we allow array sizes up to `u64`. How does the struct layout code handle huge arrays in structs? How do GCC/clang handle it?

Reply:
That's not wierd, it's wrong. Fixed.
Thanks!